### PR TITLE
Podman e2e tests

### PR DESCRIPTION
**Pull request by hidalgopl**

### CODE REVIEW

File: pkg/server/server_podman_test.go [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+
+	timeout := int(podmanAPITimeout)
+	_, err = pods.Stop(conn, api.PodName, &timeout)
+	if err != nil {
+		panic(err)
+	}
+	_, err = pods.Remove(conn, api.PodName, &forceContainerRemove)
+	if err != nil {
+		panic(err)
+	}
```
Comment (ldx) [inline pkg/server/server_podman_test.go:85]: Shouldn't we be able to remove containers and the pod via sending a pod update with no containers?

File: pkg/server/server_podman_test.go [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+	select {
+	case f := <-ws.ReadMsg():
+		c, m, err := wsstream.UnpackMessage(f)
+		assert.NoError(t, err)
+		assert.Equal(t, 1, c)
+		assert.True(t, strings.HasSuffix(string(m), "pong"))
+	case <-time.After(timeout):
+		assert.FailNow(t, "reading timed out")
+	}
+}
```
Comment (ldx) [inline pkg/server/server_podman_test.go:343]: Why separate E2E tests depending on the runtime? These tests would be useful for any runtime.

File: scripts/run_tests.sh [hunk 3]
```diff
@@ -19,10 +19,14 @@ set -e
...
 ROOT_DIR=$SCRIPT_DIR/..
+GO_EXECUTABLE=$(which go)
 
 cd $ROOT_DIR
 make
-go test ./...
+$GO_EXECUTABLE test ./...
+
+echo "running podman e2e-tests"
+sudo $GO_EXECUTABLE test ./pkg/server/   -v -args -podman=true
```
Comment (ldx) [inline scripts/run_tests.sh:29]: Why not let the caller of the script use sudo or set the right capabilities instead?

File: pkg/server/server_podman_test.go (hunk 2 above)
Comment (hidalgopl) [reply pkg/server/server_podman_test.go:343]: We have similar tests for itzo runtime in `server_test.go`, they run if `-functional` flag is set. Those here run if `-podman` flag is set. I split them into two separate files, just to have clear line which is run with either flag.

File: pkg/convert/convert.go [hunk 4]
```diff
@@ -50,6 +52,21 @@ func VolumeToK8sVolume(volume api.Volume) v1.Volume {
...
 }
 
+func handlePackagePathVolumes(volume api.Volume) v1.Volume {
+	if hostType, ok := specialVolumesTypes[volume.Name]; ok {
+		return v1.Volume{
+			Name: volume.Name,
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Type: &hostType,
+					Path: specialVolumesPaths[volume.Name],
```
[the remark this answers is not in the document]
Comment (hidalgopl) [reply pkg/convert/convert.go]: Yeap, it's fixed already, added special handling for PackagePath type in convert function copied from KIP.

File: pkg/runtime/podman.go [hunk 5]
```diff
@@ -61,6 +63,17 @@ func (ps *PodmanSandbox) RunPodSandbox(spec *api.PodSpec) error {
 	// those two are important as we deploy and mount resolv.conf and /etc/hosts
 	podSpec.NoManageResolvConf = true
 	podSpec.NoManageHosts = true
+	portMappings := make([]specgen.PortMapping, 0)
+	for _, unit := range spec.Units {
+		for _, port := range unit.Ports {
+			portMappings = append(portMappings, specgen.PortMapping{
+				ContainerPort: uint16(port.ContainerPort),
+				HostPort:      uint16(port.HostPort),
+				Protocol:      string(port.Protocol),
```
[the remark this answers is not in the document]
Comment (hidalgopl) [reply pkg/runtime/podman.go:72]: addressed in #113 , thanks :+1:

File: pkg/runtime/podman.go [hunk 6]
```diff
@@ -157,6 +154,7 @@ func (pcs *PodmanContainerService) CreateContainer(unit api.Unit, spec *api.PodS
 	containerSpec := specgen.NewSpecGenerator(container.Image, false)
 	containerSpec.Name = convert.UnitNameToContainerName(unit.Name)
 	containerSpec.Pod = api.PodName
+	containerSpec.Terminal = true
```
[the remark this answers is not in the document]
Comment (hidalgopl) [reply pkg/runtime/podman.go:153]: that was weird leftover, fixed in #113 :+1:

File: pkg/server/server.go [hunk 7]
```diff
@@ -178,13 +178,17 @@ func (s *Server) startNetworkAgent(IP, nodeName string) {
 func (s *Server) updateHandler(w http.ResponseWriter, r *http.Request) {
 	switch r.Method {
 	case "POST":
+		glog.Infof("got update request")
```
[the remark this answers is not in the document]
Comment (hidalgopl) [reply pkg/server/server.go:181]: this was actually debug leftover, thanks for catching this, fixed in #113  :+1:

**Pull request merged**